src/goto-symex/complexity_violation.h

```diff
@@ -0,0 +1,24 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_SYMEX_COMPLEXITY_VIOLATION_H
+#define CPROVER_GOTO_SYMEX_COMPLEXITY_VIOLATION_H
+
+/// What sort of symex-complexity violation has taken place.
+///
+/// Loop: If we've violated a loop complexity boundary, or are part of
+/// a blacklisted loop.
+/// Branch: If this particular branch has been deemed too complex.
+enum complexity_violationt
```

Comment:
use `enum class` (strong enum, not implicitly convertable to int / other enum, and uses `complexity_violationt::` as a name prefix rather than polluting the global namespace)